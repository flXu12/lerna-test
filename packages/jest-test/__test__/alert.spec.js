import { mount, shallowMount } from '@vue/test-utils'
import Alert from '../src/alert.vue'

//describe： function，测试套件，表示一组相关的测试
//test： function，测试用例
describe('Alert test', () => {
  test('create', () => {
    const wrapper = mount(Alert, {
      propsData: {
        title: 'I am a title',
        showIcon: true,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    //find选中匹配器的第一个DOM节点或wrapper，text()获取其中的内容
    expect(wrapper.find('.el-alert__title').text()).toEqual('I am a title')
    //通过.vm获取当前实力对象，相当于拿到了vue组件里的this对象
    expect(wrapper.vm.visible).toBe(true)
  })

  // methods
  test('close', () => {
    const wrapper = shallowMount(Alert, () => {
      propsData: {
        title: 'I am a title title title'
      }
    })
    wrapper.find('.el-alert__closebtn').trigger('click')
    expect(wrapper.vm.visible).toBe(false)
  })
})
