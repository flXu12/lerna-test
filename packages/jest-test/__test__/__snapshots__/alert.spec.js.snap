// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert test create 1`] = `
<div role="alert" class="el-alert el-alert--info is-light"><i class="el-alert__icon el-icon-info"></i>
  <div class="el-alert__content"><span class="el-alert__title">I am a title</span>
    <!---->
    <!----> <i class="el-alert__closebtn el-icon-close"></i></div>
</div>
`;
